### PR TITLE
94: Add Tabs and TabsItem components

### LINKED ISSUE #94: Add Tabs component

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
# Done
- Added [Tabs](https://docs.vanillaframework.io/en/patterns/tabs) component and TabsItem subcomponent
- Added stories/knobs to Storybook
- Added snapshot tests

# QA
- Pull code
- Run `yarn lint` and `yarn test` to ensure there are no linting or testing errors
- Run `./run serve --watch` and navigate to http://localhost:8102/
- Verify the Tabs component in Storybook matches the [Vanilla docs](https://docs.vanillaframework.io/en/patterns/tabs)

# Fixes
Fixes #94

### CODE REVIEW

File: src/lib/components/Tabs/TabsItem.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    const {
+      children, href, index, isSelected,
+    } = this.props;
+
+    return (
+      <li className="p-tabs__item" role="presentation">
+        <a
+          href={href}
+          className="p-tabs__link"
+          tabIndex={isSelected ? 0 : -1}
```
Comment (anthonydillon) [inline src/lib/components/Tabs/TabsItem.js]: Can you remove this tabIndex as -1 stops a user from tabbing to it.

**Review by anthonydillon — CHANGES_REQUESTED** [review]
One small comment about the use of tabIndex

**Comment by Caleb-Ellis** [comment]
@anthonydillon I've changed tabIndex to be index + 1 so you can still tab through in order, and added an onKeyPress func so you can navigate the tabs via keyboard

**Review by anthonydillon — APPROVED** [review]
Code looks good

**Pull request merged**